app/src/components/blocks/_lloydGeorge/lloydGeorgeSelectSearchResults/LloydGeorgeSelectSearchResults.test.tsx

```diff
@@ -109,7 +110,125 @@ describe('LloydGeorgeSelectSearchResults', () => {
+                    setSelectedDocuments: mockSetSelectedDocuments,
+                };
+
+                rerender(<LloydGeorgeSelectSearchResults {...props} />);
+
+                checkboxes.forEach((checkbox) => {
+                    expect(checkbox).toBeChecked();
+                });
+
+                expect(toggleSelectAllBtn.textContent).toBe('Deselect all files');
+            });
+
+            it('check all checkboxes unchecked, when select all files button clicked, all files previously checked', () => {
+                let selectedDocuments = mockAllSelectedDocuments;
+                mockSetSelectedDocuments.mockImplementation(
+                    (documents) => (selectedDocuments = documents),
+                );
+
+                const { rerender } = renderComponent({
+                    selectedDocuments: mockAllSelectedDocuments,
+                });
+                const toggleSelectAllBtn = screen.getByTestId('toggle-selection-btn');
+                const checkboxes = screen.getAllByRole('checkbox');
+
+                checkboxes.forEach((checkbox) => {
+                    expect(checkbox).toBeChecked();
+                });
+
+                expect(toggleSelectAllBtn.textContent).toBe('Deselect all files');
+
+                act(() => {
+                    userEvent.click(toggleSelectAllBtn);
+                });
+
+                expect(mockSetSelectedDocuments).toBeCalledWith([]);
+
+                const props: Props = {
+                    searchResults: searchResults,
+                    setSubmissionSearchState: mockSetSubmissionSearchState,
+                    selectedDocuments: selectedDocuments,
+                    setSelectedDocuments: mockSetSelectedDocuments,
+                };
+
+                rerender(<LloydGeorgeSelectSearchResults {...props} />);
+
+                checkboxes.forEach((checkbox) => {
+                    expect(checkbox).not.toBeChecked();
+                });
+                expect(toggleSelectAllBtn.textContent).toBe('Select all files');
+            });
+
+            it('check all checkboxes unchecked, when select all files button clicked twice, no files previously checked', () => {
+                let selectedDocuments: Array<string> = [];
+                mockSetSelectedDocuments.mockImplementation(
+                    (documents) => (selectedDocuments = documents),
+                );
+
+                const { rerender } = renderComponent({ selectedDocuments: selectedDocuments });
+                const toggleSelectAllBtn = screen.getByTestId('toggle-selection-btn');
+                const checkboxes = screen.getAllByRole('checkbox');
+
+                act(() => {
+                    userEvent.click(toggleSelectAllBtn);
+                });
+
+                let props: Props = {
+                    searchResults: searchResults,
+                    setSubmissionSearchState: mockSetSubmissionSearchState,
```

Comment:
for this test, I think we could shorten it a bit like below:
(but up to you if prefer the current one)

```
            it('check all checkboxes unchecked, when select all files button clicked twice, no files previously checked', () => {
                const props: Props = {
                    searchResults: searchResults,
                    setSubmissionSearchState: mockSetSubmissionSearchState,
                    selectedDocuments: [],
                    setSelectedDocuments: mockSetSelectedDocuments,
                };
                mockSetSelectedDocuments.mockImplementation(
                    (documents) => (props.selectedDocuments = documents),
                );

                const { rerender } = renderComponent(props);
                const toggleSelectAllBtn = screen.getByTestId('toggle-selection-btn');
                const checkboxes = screen.getAllByRole('checkbox');

                act(() => {
                    userEvent.click(toggleSelectAllBtn);
                });

                rerender(<LloydGeorgeSelectSearchResults {...props} />);

                act(() => {
                    userEvent.click(toggleSelectAllBtn);
                });

                expect(mockSetSelectedDocuments).toBeCalledWith([]);

                rerender(<LloydGeorgeSelectSearchResults {...props} />);

                checkboxes.forEach((checkbox) => {
                    expect(checkbox).not.toBeChecked();
                });
                expect(toggleSelectAllBtn.textContent).toBe('Select all files');
            });
```